crypto/codec/tm.go

```diff
@@ -0,0 +1,31 @@
+package codec
+
+import (
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+	tmed25519 "github.com/tendermint/tendermint/crypto/ed25519"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// FromTmPubKey converts TM's tmcrypto.PubKey to our own PubKey.
+func FromTmPubKey(tmPk tmcrypto.PubKey) (cryptotypes.PubKey, error) {
+	switch tmPk := tmPk.(type) {
+	case tmed25519.PubKey:
+		return &ed25519.PubKey{Key: []byte(tmPk)}, nil
+	default:
```

Comment:
```suggestion
case tmsecp256k1.PubKey:
		return &secp256k1.PubKey{Key: []byte(tmPk)}, nil
		default:
```

secp was added in rc6

Reply:
to keep things nice and clean (this PR is getting big already), i created a separate one https://github.com/cosmos/cosmos-sdk/pull/7838

Reply:
ah okay, thank you. Should this turn the pubkey straight into the tendermint proto public key? 

does the tendermint pubkey get used anywhere in the sdk that makes it needed?

Reply:
That's a good question, and I was hesitating about this one. In the SDK, we use both:
- `tmcrypto.PubKey` is used mainly with `tmtypes.Validator`
- `tendermint/proto/tendermint/crypto` is used in `abci.ValidatorUpdates`

I'm happy for this function to turn it directly to a `tmprotocrypto.PublicKey`, but we would still need to import tendermint/crypto/encoding everywhere we we use `tmtypes.Validator`.

Reply:
oh I didnt know tmtypes.validator is used in the sdk other than sending updates to tm. maybe its fine for now and if the work to decouple that happens this can be done then.

Reply:
I actually changed this function to return proto PublicKey, in `To/FromTmPublicKey`. Although I think we should keep `To/FromTmPubKey`  too, it's quite useful to avoid importing `tendermint/crypto/encoding` a bit everywhere.

Reply:
Good question!